### ISSUE TITLE
Support for aggregate?

### ISSUE THREAD

**Issue by Nocklas** [Automattic/monk#56]
Are there any plans to add support for aggregation to monk?

**Comment by oO** [comment]
I'm also interested in aggregate support. with now I'm monkey patching my collections with this:

	// Extend to allow aggregation. TODO: hack
	assets.aggregate = function(aggregation){
		var collection = this.col;
		var options = {};
		return function (callback){
			return collection.aggregate(aggregation, options, callback)
		}
	}

**Comment by bradoyler** [comment]
:+1:

**Comment by marcusoftnet** [comment]
That works... but why not include that functionality in Monk?

**Comment by travisjeffery** [comment]
yeah i'd like to get this in, can do it sometime this week probably

**Comment by innocentio** [comment]
Any progress on adding aggregate?

**Comment by innocentio** [comment]
Answering my own question but I went down the rabbit hole and found out there is actually dependency issue that has to be resolved first. There is currently no cursor support in node-mongoskin which monk depends on. And for the reference, those guys are not maintaining the package and so there will be no fix coming any time soon I guess. 

 I guess @oO hack-fix will have to do for now.

**Comment by heartnetkung** [comment]
I've managed to implement the support for aggregation. Just in case anyone stumble upon this topic like I did.
https://github.com/heartnetkung/monkii/commit/e80f99534fc611cb1f0b7a7d87f9ce715419f0ed

**Comment by indieforger** [comment]
@heartnetkung Could you perhaps write test for it as well and do pull request? It would be nice to have that feature in.

**Comment by heartnetkung** [comment]
@indieforger I'm thinking of sending a pull request as well, but I think this repo is deserted. This repo has lots of problems like old dependencies and bson and node v5. Should I move to this [fork](https://github.com/ratson/monkii) instead? Of course, moving to mongodb 2.x requires a different effort from api change and all.

ps. I'm quite new to opensource and willing to help. Not really sure what to do.

**Comment by indieforger** [comment]
@heartnetkung Same here, when it comes down to contributing to open-source. Best way would probably to contact the lead maintainer to figure out what's the best course of action. I think you could ping @Automatic or @rauchg, as he contributed the most.

Are you on twitter?  If yes you can ping me on [indieforger](https://twitter.com/indieforger) or [fullstackforger](https://twitter.com/indieforger).

**Comment by heartnetkung** [comment]
@indieforger Thanks for the response. Here's my [twitter](https://twitter.com/heartnetkung).

About the issue, I managed to write all the test cases with `aggregate` and got them to work properly with my project. Additionally, I also did some bug fixes related to the new dependencies so that it would work in production. Here's the [git url](https://github.com/heartnetkung/monkii) for all the fixes. I'll come back for the proper pull request in this repo later when I have the time.
